aiosmtpd/tests/test_smtp.py

```diff
@@ -984,29 +1064,75 @@ def test_ehlo(self):
             code, response = client.ehlo('example.com')
             self.assertEqual(code, 250)
             lines = response.splitlines()
-            expecteds = (
+            expecteds = [
                 bytes(socket.getfqdn(), 'utf-8'),
                 b'SIZE 33554432',
                 b'SMTPUTF8',
-                b'AUTH LOGIN NULL PLAIN WITH-DASH WITH-MULTI-DASH WITH_UNDERSCORE',
+                (
+                    b'AUTH DENYFALSE DENYMISSING LOGIN NONE NULL PLAIN '
+                    b'WITH-DASH WITH-MULTI-DASH WITH_UNDERSCORE'
+                ),
                 b'HELP',
-            )
-            for actual, expected in zip(lines, expecteds):
-                self.assertEqual(actual, expected)
+            ]
+            assert lines == expecteds
```

Comment:
Again, do direct compare list against list. The line-by-line comparison sometimes resulted in confusing failures. By comparing whole list against whole list, I can get a better grasp of what's going on.

Also, transition to pytest's `assert`-based testing.